feat: Pixelate an image

viveknathani opened the pull request:
Pixelating an image involves making square sections in your image
that have all their pixel values set to a common value. In our case,
we set all the values in a section/submatrix/window to the average
pixel value in that area.

viveknathani committed:
feat: Pixelate an image

Pixelating an image involves making square sections in your image
that have all their pixel values set to a common value. In our case,
we set all the values in a section/submatrix/window to the average
pixel value in that area.

priyanshul-govil reviewed (changes requested):
1. `pixelRatio` does not seem a good name for the purpose it serves. *pixelWidth*, *windowWidth* might be clearer.
2. It's better to have more variables denoting *totalRed*, *totalGreen*, *totalBlue* as lines 482-484 don't make logical sense on their own. It'll be better to calculate lines 490-492 as `avgRed = totalRed / count` and so on.

viveknathani committed:
refactor: Unsuitable naming and computing

'pixelRatio' is changed to 'pixelWidth' to provide a name that suits
better. Average calculation could be better by using three more
variables to handle the summation.

viveknathani commented:
Done

priyanshul-govil commented on src/secur3dit/filters/Filters.java:
```diff
@@ -463,41 +463,41 @@ public static BufferedImage pixelate(BufferedImage image, int pixelRatio) {
         int avgGreen = 0;
```
variables should be declared in the scope they are used.

priyanshul-govil reviewed (changes requested):
*average* variables have been declared a scope above which they are meant to be used.

viveknathani committed:
refactor: above the scope declare, needless check

The scope of average variables is changed to where they should exist.
Checking the indices for apt submatrix region was not needed since the
values just increment by pixelWidth, always being divisible by
pixelWidth.

viveknathani commented:
Took care of that. And also removed a needless check for indices. Details are mentioned in the commit body.

viveknathani committed:
refactor: outdated comments, whitespace near '/'

Removed an outdated comment. Changed an exisiting comment to make it
more explicit. Whitespace is needed before and after binary operators.

priyanshul-govil reviewed (approved):
seems good now.
